### PR TITLE
Add RBAC rules and basic service account

### PR DESCRIPTION
**Pull request by sbreker**
Adds a named service account and restricts access to retrieving pod info only.

### CODE REVIEW

**Comment by scollazo** [comment]
I'm getting an error when applying this:

 ```error: error validating "serviceaccount.yaml": error validating data: ValidationError(ServiceAccount.metadata): unknown field "automountServiceAccountToken" in io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta; if you choose to ignore these errors, turn validation off with --validate=false```

I'm using :
```
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6+k3s1", GitCommit:"a8e0c66d1a90a2bbc4ffa975129ca35756cc7c14", GitTreeState:"clean", BuildDate:"2022-09-28T16:52:07Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6+k3s1", GitCommit:"a8e0c66d1a90a2bbc4ffa975129ca35756cc7c14", GitTreeState:"clean", BuildDate:"2022-09-28T16:52:07Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
```

File: hack/kube/base/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: sdps-serviceaccount
+  automountServiceAccountToken: false
```
Comment (scollazo) [inline hack/kube/base/serviceaccount.yaml]: This should be at metadata level, as per https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#opt-out-of-api-credential-automounting

**Review by scollazo — CHANGES_REQUESTED** [review]

File: hack/kube/base/serviceaccount.yaml (hunk 1 above)
Comment (sbreker) [reply hack/kube/base/serviceaccount.yaml]: This is fixed.

**Review by scollazo — APPROVED** [review]
LGTM

**Pull request merged**